pogs/interface_test.go

```diff
@@ -0,0 +1,168 @@
+package pogs
+
+import (
+	"golang.org/x/net/context"
+	"testing"
+	"zombiezen.com/go/capnproto2"
+	air "zombiezen.com/go/capnproto2/internal/aircraftlib"
+)
+
+type simpleEcho struct{}
+
+func checkFatal(t *testing.T, name string, err error) {
+	if err != nil {
+		t.Fatalf("%s for TestInsertIFace: %v", name, err)
+	}
+}
+
+func (s simpleEcho) Echo(p air.Echo_echo) error {
+	text, err := p.Params.In()
+	if err != nil {
+		return err
+	}
+	p.Results.SetOut(text)
+	return nil
+}
+
+type EchoBase struct {
+	Echo air.Echo
+}
+
+type EchoBases struct {
+	Bases []EchoBase
+}
+
+type Hoth struct {
+	Base EchoBase
+}
+
+func TestInsertIFace(t *testing.T) {
```

Comment:
Using reflect.DeepEqual on interfaces doesn't really do what we want in this case, and it seemed prudent to have something that tested the behavior of the interface, which would require special casing in the general test anyway. I think at the time I decided making it work would be more disruptive to the existing tests than really made sense. I'm willing to be convinced otherwise.